internals/overlord/logstate/opentelemetry/opentelemetry_test.go

```diff
@@ -0,0 +1,385 @@
+		Message: "log line #9\n",
+	}}
+
+	expected := compactJSON(`
+{"resourceLogs": [
+	{
+		"resource": {
+			"attributes": [{"key": "service.name", "value": {"stringValue": "svc1"}}]
+		},
+		"scope_logs": [{
+			"scope": {"name": "pebble", "version": "v1.23.0-dev"},
+			"log_records": [
+				{
+					"timeUnixNano": 1704026090000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #1"}
+				},
+				{
+					"timeUnixNano": 1704026092000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #3"}
+				},
+				{
+					"timeUnixNano": 1704026094000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #5"}
+				},
+				{
+					"timeUnixNano": 1704026097000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #8"}
+				}
+			]
+		}]
+	},{
+		"resource": {
+			"attributes": [{"key": "service.name", "value": {"stringValue": "svc2"}}]
+		},
+		"scope_logs": [{
+			"scope": {"name": "pebble", "version": "v1.23.0-dev"},
+			"log_records": [
+				{
+					"timeUnixNano": 1704026091000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #2"}
+				},
+				{
+					"timeUnixNano": 1704026096000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #7"}
+				}
+			]
+		}]
+	},{
+		"resource": {
+			"attributes": [{"key": "service.name", "value": {"stringValue": "svc3"}}]
+		},
+		"scope_logs": [{
+			"scope": {"name": "pebble", "version": "v1.23.0-dev"},
+			"log_records": [
+				{
+					"timeUnixNano": 1704026093000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #4"}
+				},
+				{
+					"timeUnixNano": 1704026095000000000,
+					"severityText": "",
+					"severityNumber": 0,
+					"body": {"stringValue": "log line #6"}
+				}
+			]
+		}]
+	},{
+		"resource": {
+			"attributes": [{"key": "service.name", "value": {"stringValue": "svc4"}}]
+		},
+		"scope_logs": [{
+			"scope": {"name": "pebble", "version": "v1.23.0-dev"},
+			"log_records": [{
+				"timeUnixNano": 1704026098000000000,
+				"severityText": "",
+				"severityNumber": 0,
+				"body": {"stringValue": "log line #9"}
+			}]
+		}]
+	}
+]}`)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		c.Assert(r.Method, Equals, http.MethodPost)
+		c.Assert(r.Header.Get("Content-Type"), Equals, "application/json")
+
+		reqBody, err := io.ReadAll(r.Body)
+		c.Assert(err, IsNil)
+		c.Assert(string(reqBody), DeepEquals, string(expected))
+	}))
+	defer server.Close()
+
+	client := opentelemetry.NewClient(&plan.LogTarget{Location: server.URL})
+	client.SetLabels("svc1", map[string]string{})
+	client.SetLabels("svc2", map[string]string{})
+	client.SetLabels("svc3", map[string]string{})
+	client.SetLabels("svc4", map[string]string{})
```

Comment:
I think if we're going to have a separate `TestLabels` we should drop these calls. But we could just call `SetLabels` with a couple of labels on (say) svc4 and combine the two tests. Most of it will be the same.